backend/asmgen.ml

```diff
@@ -488,6 +488,11 @@ let build_asm_directives () : (module Asm_targets.Asm_directives_intf.S) = (
   )
 
 let emit_begin_assembly_with_dwarf ~disable_dwarf ~emit_begin_assembly ~sourcefile () =
+  if !Flambda_backend_flags.internal_assembler then
+    (X86_proc.register_internal_assembler Internal_assembler.assemble;
+    Emitaux.binary_backend_available := true;
+    Emitaux.create_asm_file := false)
```

Comment:
Can you check that `-S` still works and leaves behind a `.s` file even when the internal assembler is being used?  In this case it should still invoke the internal assembler to do the actual assembling.

Reply:
Fixed